docs/source/performance/cache-backends.mdx

```diff
@@ -0,0 +1,80 @@
+
+Unfortunately, the default configuration of Apollo Server exposes you to denial of service attacks.
+This is because APQs are enabled by default and the default cache in Apollo Server 3 is unbounded, meaning an attacker can exhaust your memory and crash your server. The default cache in Apollo Server 4 will be bounded; **we recommend opting in to that behavior by providing the `cache: "bounded"` option to your Apollo Server constructor or configuring the `cache` yourself.**
+
+## Configuring in-memory caching
+
+By default, Apollo Server's caching features use an unbounded cache. This is not safe for production use. If you'd like to configure the in-memory cache, Apollo provides the [`InMemoryLRUCache`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#inmemorylrucache) class from the `@apollo/utils.keyvaluecache` package. This class is a wrapper around the `lru-cache` package with a default maximum of approximately 30MiB of memory. The `InMemoryLRUCache` is configurable with the options from `lru-cache`. See the [`lru-cache` documentation`](https://www.npmjs.com/package/lru-cache) for more information on configuration options.
+
+### Example: using `InMemoryLRUCache`
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // Note: this is equivalent to providing `cache: "bounded"`
+  cache: new InMemoryLRUCache(),
+});
+```
+
+### Example: configuring `InMemoryLRUCache`
+
+In this example, we've increased the default size and provided a default TTL. For more information on the options, see the [`lru-cache` documentation`](https://www.npmjs.com/package/lru-cache).
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // ...
+  cache: new InMemoryLRUCache({
+    // ~100MiB
+    maxSize: Math.pow(2, 20) * 100,
+    // 5 minutes
+    ttl: 300000,
+  }),
+});
+```
+
+## Configuring external caching
+
+Apollo no longer maintains any caching backends directly. Instead, we recommend using the [`keyv`](https://www.npmjs.com/package/keyv) package along with the [`KeyvAdapter`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyvAdapter#keyvadapter-class) class provided by the `@apollo/utils.keyvadapter` package. `KeyvAdapter` simply wraps a `Keyv` instance and implements the `KeyValueCache` interface which is required by Apollo Server.
+
+### Example: using `Keyv` with Redis
+
+```ts
+import Keyv from 'keyv';
+import { KeyvAdapter } from '@apollo/utils.keyvadapter';
+
+const server = new ApolloServer({
+  // ...,
+  cache: new KeyvAdapter(new Keyv('redis://localhost:6379')),
+});
+```
+
+## Implementing your own cache backend
+
+If your requirements are more specialized or you prefer to implement your own cache backend, you can implement the `KeyValueCache` interface and pass it to Apollo Server's constructor.
+
+The interface is very simple:
```

Comment:
I feel like clarifying what the below snippet is showing is helpful. So, something like: 
```suggestion
The `KeyValueCache` interface is shown below:
```